brooklyn-tosca-transformer/src/main/java/io/cloudsoft/tosca/a4c/brooklyn/spec/VanillaSoftwareProcessLifecycleModifier.java

```diff
@@ -65,96 +74,94 @@ public void apply(EntitySpec<?> entitySpec, NodeTemplate nodeTemplate, Topology
-                    script = new ResourceUtils(this)
-                            .getResourceAsString(csarPath.getParent().toString() + EXPANDED_FOLDER + ref);
-                } catch (CSARVersionNotFoundException | NullPointerException e) {
-                    script = new ResourceUtils(this).getResourceAsString(ref);
-                }
-
-                Map<String, PaaSNodeTemplate> builtPaaSNodeTemplates = treeBuilder.buildPaaSTopology(topology).getAllNodes();
-                String computeName = (nodeTemplate.getName() != null) ? nodeTemplate.getName()
-                                                                      : (String) spec.getFlags().get("tosca.template.id");
-                PaaSNodeTemplate paasNodeTemplate = builtPaaSNodeTemplates.get(computeName);
-
-                StringBuilder inputBuilder = new StringBuilder();
-                Map<String, IValue> inputParameters = op.getInputParameters();
-                if (inputParameters != null) {
-                    for (Map.Entry<String, IValue> entry : inputParameters.entrySet()) {
-                        // case keyword SOURCE used on a NodeType
-                        Optional<Object> value = resolve(inputParameters, entry.getKey(), paasNodeTemplate, builtPaaSNodeTemplates);
-                        inputBuilder.append(String.format("export %s=%s%n", entry.getKey(), value.or("")));
-                    }
-                }
-
-                spec.configure(cmdKey, inputBuilder.toString() + "\n" + script);
-            } else {
-                LOG.warn("Unsupported operation implementation for " + opKey + ": " + artifact + " has no ref");
+
+        String script = getScript(artifact.getArchiveName(), artifact.getArchiveVersion(), ref);
+
+        Map<String, PaaSNodeTemplate> builtPaaSNodeTemplates = treeBuilder.buildPaaSTopology(topology).getAllNodes();
+        String computeName = (nodeTemplate.getName() != null) ? nodeTemplate.getName() : (String) spec.getFlags().get("tosca.template.id");
+        PaaSNodeTemplate paasNodeTemplate = builtPaaSNodeTemplates.get(computeName);
+        spec.configure(cmdKey, buildExportStatements(op, paasNodeTemplate, builtPaaSNodeTemplates) + "\n" + script);
+    }
+
+    private String buildExportStatements(Operation op, PaaSNodeTemplate paasNodeTemplate, Map<String, PaaSNodeTemplate> builtPaaSNodeTemplates) {
+        StringBuilder inputBuilder = new StringBuilder();
+        Map<String, IValue> inputParameters = op.getInputParameters();
+        if (inputParameters != null) {
+            for (Map.Entry<String, IValue> entry : inputParameters.entrySet()) {
+                // case keyword SOURCE used on a NodeType
+                Optional<Object> value = resolve(inputParameters, entry.getKey(), paasNodeTemplate, builtPaaSNodeTemplates);
+                inputBuilder.append(String.format("export %s=%s%n", entry.getKey(), value.or("")));
```

Comment:
Ah I think I understand you now, if we use this platform's line separator it may differ from the platform the script is run on!  I presume `\n` is safest then since it is a bash script?